code/game/objects/items/weapons/material/shards.dm

```diff
@@ -61,6 +61,63 @@
 			return
 	return ..()
 
+/obj/item/weapon/material/shard/afterattack(var/atom/target, mob/living/carbon/human/user as mob)
+	var/active_hand //hand the shard is in
+	var/will_break
+	var/gloves_are_heavy = FALSE//this is a fucking mess
+	var/break_damage = 4
+	var/light_glove_d = rand(2, 4)
+	var/no_glove_d = rand(4, 6)
+	var/list/h_gloves = list(/obj/item/clothing/gloves/captain, /obj/item/clothing/gloves/cyborg,
+							/obj/item/clothing/gloves/swat, /obj/item/clothing/gloves/combat,
+							/obj/item/clothing/gloves/botanic_leather, /obj/item/clothing/gloves/duty,
+							/obj/item/clothing/gloves/tactical, /obj/item/clothing/gloves/vox,
+							/obj/item/clothing/gloves/gauntlets)
+
+	if(istype(user.l_hand, src))
+		active_hand = BP_L_HAND
+	else
+		active_hand = BP_R_HAND
+
+	if(prob(75))
+		will_break = TRUE
+	else
+		will_break = FALSE
+
+	if(user.gloves && (user.gloves.body_parts_covered & HANDS))
+		var/obj/item/clothing/gloves/UG = user.gloves
+		user << "[UG]"
+		for(var/I in h_gloves)
+			user << "[I]"
+			if(istype(I, user.gloves)) //if gloves are heavy

```

Comment:
this is not working for some reason; i think it's because during testing, i found out that 'user.gloves' outputs the name of the gloves instead of the type of the gloves. don't know how to get the type of the gloves.

Reply:
`if(user.gloves && istype(I, user.gloves.type))`
First part not necessary if you've already enforced the existence of user.gloves